core/src/scene/sceneLoader.cpp

```diff
@@ -1049,6 +1024,69 @@ std::vector<StyleParam> SceneLoader::parseStyleParams(Node params, Scene& scene,
     return out;
 }
 
+StyleUniforms SceneLoader::parseStyleUniforms(const Node& value, Scene& scene) {
+    std::string type = "";
+    std::vector<UniformValue> uniformValues;
+    if (value.IsScalar()) { //float, int (bool), string (texture)
+        UniformValue uVal;
+        try {
+            uVal = value.as<float>();
+            type = "float";
+        } catch (const BadConversion& e) {
+            auto strVal = value.as<std::string>();
+            type = "int";
+            if (strVal == "true") {
+                uVal = 1;
+            } else if (strVal == "false") {
+                uVal = 0;
+            } else {
+                type = "sampler2D";
+                uVal = strVal;
+                auto texItr = scene.textures().find(strVal);
+                if (texItr == scene.textures().end()) {
```

Comment:
if a texture has not been loaded before then it will go and load it with default texture properties and add it to the list of textures maintained by the scene.